presto-main/src/main/java/com/facebook/presto/type/khyperloglog/KHyperLogLogStateFactory.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.type.khyperloglog;
+
+import com.facebook.presto.array.ObjectBigArray;
+import com.facebook.presto.spi.function.AccumulatorStateFactory;
+import com.facebook.presto.spi.function.GroupedAccumulatorState;
+import org.openjdk.jol.info.ClassLayout;
+
+public class KHyperLogLogStateFactory
+        implements AccumulatorStateFactory<KHyperLogLogState>
+{
+    private static final int SIZE_OF_SINGLE = ClassLayout.parseClass(SingleKHyperLogLogState.class).instanceSize();
+    private static final int SIZE_OF_GROUPED = ClassLayout.parseClass(GroupedKHyperLogLogState.class).instanceSize();
+
+    @Override
+    public KHyperLogLogState createSingleState()
+    {
+        return new SingleKHyperLogLogState();
+    }
+
+    @Override
+    public Class<? extends KHyperLogLogState> getSingleStateClass()
+    {
+        return SingleKHyperLogLogState.class;
+    }
+
+    @Override
+    public KHyperLogLogState createGroupedState()
+    {
+        throw new IllegalArgumentException("GROUP BY of KHyperLogLog is not supported.");
```

Comment:
Let's throw a `PrestoException(NOT_SUPPORTED`, as `IllegalArgumentException` will surface as an internal error in Presto.

Reply:
OK!